compiler/src/dmd/expressionsem.d

```diff
@@ -15732,7 +15732,15 @@ private Expression toLvalueImpl(Expression _this, Scope* sc, const(char)* action
             return visit(_this);
         }
         if (_this.isLvalue())
+        {
+            with (_this) // FIXME: avoid new
+            if (mod && !new TypePointer(e1.type).implicitConvTo(new TypePointer(to)))
```

Comment:
Is there a way to avoid allocating to get a TypePointer?

Reply:
`e1.type.pointerTo()` IIRC